metpy/interpolate/slices.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2018 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Tools for interpolating to a vertical slice/cross section through data."""
+
+import cartopy.crs as ccrs
+import numpy as np
+import xarray as xr
+
+from ..package_tools import Exporter
+from ..xarray import CFConventionHandler
+
+exporter = Exporter(globals())
+
+
+@exporter.export
+def cross_section(data, start, end, steps=100, interp_type='linear'):
```

Comment:
Thoughts on having a `points` kwarg that takes a tuple of points so we can have multiple points along a path?

Reply:
Most implementations I've seen elsewhere just take a "straight" path between two points (with straight varying in definition), so I'd lean towards keeping a simple implementation like this as an option. Also, counting out the interpolation steps between the various sub-paths could get unwieldy.

However, could an alternate solution to this be something like the below?

- pull out the interpolation part of the `cross_section` function as an `interpolate_to_slice` or `interpolate_to_path` function that takes the whole path as an argument
- pull out the geodesic construction part of the `cross_section` function (as `geodesic`?)
- have `cross_section` use these together to provide the basic functionality of taking a cross section along a geodesic between two points

This should allow someone who wants a more complicated cross section (be it multiple geodesics connecting more than two points, or some other path) to build their path, but still have a nice wrapper around the xarray (or possibly metpy) interpolation functionality.